certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -437,21 +441,37 @@ def _rank_matches_by_name_and_ssl(self, vhost_list, target_name):
             if name_type == 'exact':
                 matches.append({'vhost': vhost,
                                 'name': name,
-                                'rank': 0 if vhost.ssl else 1})
+                                'rank': NAME_RANK})
             elif name_type == 'wildcard_start':
                 matches.append({'vhost': vhost,
                                 'name': name,
-                                'rank': 2 if vhost.ssl else 3})
+                                'rank': START_WILDCARD_RANK})
             elif name_type == 'wildcard_end':
                 matches.append({'vhost': vhost,
                                 'name': name,
-                                'rank': 4 if vhost.ssl else 5})
+                                'rank': END_WILDCARD_RANK})
             elif name_type == 'regex':
                 matches.append({'vhost': vhost,
                                 'name': name,
-                                'rank': 6 if vhost.ssl else 7})
+                                'rank': REGEX_RANK})
         return sorted(matches, key=lambda x: x['rank'])
```

Comment:
The length of the name match is used to break wildcard ties in `parser.get_best_match`, but not here. Is the expectation that if it matters, you should call `_select_best_name_match` to disambiguate?

Reply:
Yes, the two functions are meant to work together. This function should never be called directly. OR at least that's what I'm assuming based on the code, which I did not originally write.

Reply:
Ok, while we're here let's add a comment in `_rank_matches_by_name` to clarify that it should be called with `_select_best_name_match`? Otherwise the PR LGTM!